File: Test Case Generation/deepseek/python/test_task_14.py
import pytest
from typing import List
from your_module import all_prefixes  # Replace with actual module name

def test_empty_string():
    """Test with empty string"""
    assert all_prefixes("") == []

def test_single_character():
    """Test with single character string"""
    assert all_prefixes("a") == ["a"]

def test_two_characters():
    """Test with two character string"""
    assert all_prefixes("ab") == ["a", "ab"]

def test_three_characters():
    """Test with three character string (docstring example)"""
    assert all_prefixes("abc") == ["a", "ab", "abc"]

def test_long_string():
    """Test with longer string"""
    assert all_prefixes("hello") == ["h", "he", "hel", "hell", "hello"]

def test_string_with_spaces():
    """Test string containing spaces"""
    assert all_prefixes("a b") == ["a", "a ", "a b"]

def test_string_with_numbers():
    """Test string containing numbers"""
    assert all_prefixes("123") == ["1", "12", "123"]

def test_string_with_special_characters():
    """Test string containing special characters"""
    assert all_prefixes("@#$") == ["@", "@#", "@#$"]

def test_unicode_string():
    """Test with unicode characters"""
    assert all_prefixes("café") == ["c", "ca", "caf", "café"]

def test_very_long_string():
    """Test with very long string"""
    long_string = "x" * 1000
    result = all_prefixes(long_string)
    assert len(result) == 1000
    assert result[0] == "x"
    assert result[-1] == long_string
    assert all(len(result[i]) == i + 1 for i in range(len(result)))

def test_string_result_length():
    """Test that number of prefixes equals string length"""
    test_strings = ["", "a", "ab", "abc", "abcd", "hello world"]
    for s in test_strings:
        result = all_prefixes(s)
        assert len(result) == len(s)

def test_prefix_incremental_building():
    """Test that each prefix builds upon the previous one"""
    test_string = "testing"
    result = all_prefixes(test_string)
    
    for i in range(1, len(result)):
        assert result[i].startswith(result[i-1])
        assert len(result[i]) == len(result[i-1]) + 1

def test_output_type():
    """Test that return type is List[str]"""
    result = all_prefixes("test")
    assert isinstance(result, list)
    assert all(isinstance(item, str) for item in result)

def test_non_mutability():
    """Test that function doesn't have side effects"""
    original = "abc"
    result = all_prefixes(original)
    assert original == "abc"
    assert result == ["a", "ab", "abc"]

def test_verify_all_prefixes_correct():
    """Test explicit verification of all prefixes for a given string"""
    test_string = "python"
    result = all_prefixes(test_string)
    expected = ["p", "py", "pyt", "pyth", "pytho", "python"]
    assert result == expected

def test_repeated_characters():
    """Test string with repeated characters"""
    assert all_prefixes("aaa") == ["a", "aa", "aaa"]

def test_mixed_case():
    """Test string with mixed case"""
    assert all_prefixes("AbC") == ["A", "Ab", "AbC"]

File: Test Case Generation/deepseek/cpp/test_task_14.cpp
#include <gtest/gtest.h>
#include <vector>
#include <string>
#include "all_prefixes.h" // Assume the function is declared in this header

using namespace std;

TEST(AllPrefixesTest, EmptyString) {
    vector<string> result = all_prefixes("");
    EXPECT_TRUE(result.empty());
}

TEST(AllPrefixesTest, SingleCharacter) {
    vector<string> result = all_prefixes("a");
    vector<string> expected = {"a"};
    EXPECT_EQ(result, expected);
}

TEST(AllPrefixesTest, ThreeCharacters) {
    vector<string> result = all_prefixes("abc");
    vector<string> expected = {"a", "ab", "abc"};
    EXPECT_EQ(result, expected);
}

TEST(AllPrefixesTest, StringWithSpaces) {
    vector<string> result = all_prefixes("hello world");
    vector<string> expected = {"h", "he", "hel", "hell", "hello", "hello ", "hello w", "hello wo", "hello wor", "hello worl", "hello world"};
    EXPECT_EQ(result, expected);
}

TEST(AllPrefixesTest, StringWithNumbers) {
    vector<string> result = all_prefixes("123");
    vector<string> expected = {"1", "12", "123"};
    EXPECT_EQ(result, expected);
}

TEST(AllPrefixesTest, StringWithSpecialCharacters) {
    vector<string> result = all_prefixes("!@#");
    vector<string> expected = {"!", "!@", "!@#"};
    EXPECT_EQ(result, expected);
}

TEST(AllPrefixesTest, LongString) {
    string longStr(100, 'x');
    vector<string> result = all_prefixes(longStr);
    EXPECT_EQ(result.size(), 100);
    for (size_t i = 0; i < 100; ++i) {
        EXPECT_EQ(result[i], string(i + 1, 'x'));
    }
}

TEST(AllPrefixesTest, UnicodeCharacters) {
    vector<string> result = all_prefixes("こんにちは");
    vector<string> expected = {"こ", "こん", "こんに", "こんにち", "こんにちは"};
    EXPECT_EQ(result, expected);
}

TEST(AllPrefixesTest, MixedCase) {
    vector<string> result = all_prefixes("AbC");
    vector<string> expected = {"A", "Ab", "AbC"};
    EXPECT_EQ(result, expected);
}

TEST(AllPrefixesTest, RepeatedCharacters) {
    vector<string> result = all_prefixes("aaaa");
    vector<string> expected = {"a", "aa", "aaa", "aaaa"};
    EXPECT_EQ(result, expected);
}